packages/react-reconciler/src/ReactFiberThrow.new.js

```diff
@@ -260,6 +263,20 @@ function throwException(
               enqueueUpdate(sourceFiber, update);
             }
           }
+          if (sourceFiber.tag === SimpleMemoComponent) {
+            const currentSourceFiber = sourceFiber.alternate;
+            if (currentSourceFiber !== null) {
+              // When we try rendering again, we should not reuse the current fiber,
+              // since it's known to be in an inconsistent state. Use a force update to
+              // prevent a bail out.
+              const update = createUpdate(NoTimestamp, SyncLane, null);
+              update.tag = ForceUpdate;
+              if (sourceFiber.updateQueue === null) {
+                sourceFiber.updateQueue = createFunctionComponentUpdateQueue();
+              }
+              ((sourceFiber.updateQueue: any): FunctionComponentUpdateQueue).hasForceUpdate = true;
```

Comment:
Should this ever get reset to false somewhere? I assume not because the update queue itself is supposed to get cleared up.